agent/resources/cgroup/cgroup_controller_linux.go

```diff
@@ -0,0 +1,78 @@
+// +build linux
+
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cgroup
+
+import (
+	"github.com/cihub/seelog"
+	"github.com/containerd/cgroups"
+	"github.com/pkg/errors"
+)
+
+// factory here is only to ease testing and it does not maintain any state
+var factory CgroupFactory = &GlobalCgroupFactory{}
+
+// Create creates a new cgroup based off the spec post validation
+func Create(cgroupSpec *Spec) (cgroups.Cgroup, error) {
```

Comment:
Can we have the [GlobalCgroupFactory](https://github.com/aws/amazon-ecs-agent/pull/917/files#diff-ab18a83c62201eea885f981fc8b94475R41) implement the `Create` and `Remove`, then we can have an obj that's responsible for creating/deleting the cgroup, also we can get rid of the global variable `CgroupFactory`? Though I don't have strong opposition of the current pattern, it just makes me feels better to have an obj to manage these process instead of calling it anywhere directly like the c style?